### ISSUE TITLE
[Bug] Rating & Sync problems with Movies library

### ISSUE THREAD

**Issue by burekas7** [razzeee/script.trakt#235]
When I watching movies (Usually from my Movies library) it doesn't updates the Trakt site automatically at the end,
I'm always need to run the Trakt's add-on manually in order to do that.
In addition, I don't get the rating popup anymore after watching movies,
- These 2 problems occur only after movies (with TV-Shows everything is OK)

I checked the settings and everything configured well:
- It configured on 75% of video watchin to enable rating.
- The option of rating Movies is 'On' (also scrobbles)
- Also "Added watched movies to Trakt" is 'On'.

Tested on official release and on Beta version 3.0.4 of Trakt

The last Log files are here (If they still available)
 Log file: Movies (Bug: No rating automatically after watching and no update to Trakt site)
http://xbmclogs.com/pghyoa6vg
 Log file: TV-Shows (Working good - There is no bug)
http://xbmclogs.com/pp66f8kd8

Forum links:
http://forum.kodi.tv/showthread.php?tid=220547&page=36
http://forum.kodi.tv/showthread.php?tid=220547&page=37

**Comment by razzeee** [comment]
Your using some kind of third party addon to view movies? Or how are they stored? Or are these salts movies?

**Comment by burekas7** [comment]
The libraries are based on Genesis folder (folders with strm files),
I'm using the same addon for watching movies and also for TV-Shows.
With TV-Shows I don't have this problem but with movies I have.
(My Trakt account is already set in Genesis settings)

**Comment by razzeee** [comment]
@tknorris Ping

For some reason there is no title and year on these items.

**Comment by tknorris** [comment]
Could that be caused by this commit: https://github.com/Razzeee/script.trakt/commit/38cd451e4c4eab22bb04e463370034a4fb1c532e

**Comment by razzeee** [comment]
No, thats me trying to fix that problem without recreating it.
That fixed the problem, but it turns out it fails later on a similar thing.

**Comment by tknorris** [comment]
In keeping with the spirit of the change you made to avoid having the addon have to include the title and year with video_ids, I think we can fix it by populating the title and year in preFetchUserRatings() since by that time we've made the summary and/or scrobble trakt calls and now have the data from trakt based on the video_ids.

I'll make the change either today or tomorrow and test it out.

**Comment by vonH** [comment]
I've not seen this problem - I recently had to synchronise the status for 870 movies, it took a long time (as expected) but worked well.

**Comment by tknorris** [comment]
Sync works fine. It's ID scrobbling for movies that's the issue.

**Comment by s7eele** [comment]
@tknorris i may be misunderstanding something but this looked odd to me as well, I just tested two movies that were both added to my library via genesis (.strm) and i got a successful 'real-time' scrobble, Mark as watched and rating on both, just looked odd so chiming in, can provide log if needed.

edit - perhaps scraper issue, shouldve looked at log I guess :/

**Comment by razzeee** [comment]
@burekas7 
are you sure that the strm files are scraped?

**Comment by tknorris** [comment]
According to the log playback is directly from the add-on.

**Comment by s7eele** [comment]
checked that too, got good scrobble, mark as watched and rating straight from genesis also, not seeing a lot in log, looked like 'the maze runner' scraped okay but...

**Comment by tknorris** [comment]
You'll only see the problem if all of the following are true: 1) It's a movie and 2) scrobble is turned on and 3) ratings are turned on and 4) playback is from an addon and 5) the addon sets the script.trakt.ids attribute and 6a) you are running 3.0.4~beta1 or 6b) You are running 3.0.3 and the addon doesn't pass title and year.

**Comment by s7eele** [comment]
deleted

**Comment by tknorris** [comment]
https://github.com/Razzeee/script.trakt/pull/212 submitted to fix this issue. It would be good if someone could do a github install to test this. @Razzeee let me know if you want me to push a beta once this has been tested.

**Comment by razzeee** [comment]
@burekas7 
can you try with https://github.com/tknorris/script.trakt/tree/dev and report back?

**Comment by burekas7** [comment]
Thanks guys!!! @Razzeee @tknorris @s7eele 
Finally it works also with the Movies library.
Could you make an official release like the dev version?

One more question: Is Trakt supports add-ons such as Pulsar etc?

By the way, there is very annoying bug with the pin code on each fresh installation (I also wrote it in the forum)
To write it in the forum or open it as a new issue here?

**Comment by razzeee** [comment]
New issue here please
